sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -447,15 +447,23 @@ kernel_bundle(kernel_bundle<State> &&) -> kernel_bundle<State>;
 namespace detail {
 // Internal non-template versions of get_kernel_id API which is used by public
 // onces
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+__SYCL_EXPORT kernel_id get_kernel_id_impl(string_view KernelName);
+#else
 __SYCL_EXPORT kernel_id get_kernel_id_impl(std::string KernelName);
+#endif
 } // namespace detail
 
 /// \returns the kernel_id associated with the KernelName
 template <typename KernelName> kernel_id get_kernel_id() {
   // FIXME: This must fail at link-time if KernelName not in any available
   // translation units.
   using KI = sycl::detail::KernelInfo<KernelName>;
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  return detail::get_kernel_id_impl(detail::string_view(KI::getName()));
+#else
```

Comment:
We have to provide old interface symbols, but we're not obligated to use them. You can make `string_view` overload always available and use it unconditionally here.

Reply:
Done. Thanks.

Reply:
I removed #ifdef, #else, and #endif to make both old and new APIs available unconditionally.
https://github.com/intel/llvm/pull/12193/commits/d62e1755bd2bcad40e70351c35e50697e95795d5
Then, it is caught by  the pre-commit test.
```
Failed Tests (1):
  SYCL :: abi/sycl_symbols_linux.dump
```
I did not know that **adding** a new method is also an ABI symbols list change.
I will have to revert this change.

Reply:
It's an ABI change. but not a breaking one, as mentioned in the test failure message that you got. You should be able to just update the test.

Reply:
I thought we were not allowed to change the ABI symbol list without changing the version number, which is not allowed in this period.
I can change the list of ABI symbols.